File: hw/jacome/U3/HW22Strategy/HW22Strategy/src/main/java/ec/edu/espe/hw22strategy/view/SortApp.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package ec.edu.espe.hw22strategy.view;

import ec.edu.espe.hw22strategy.controller.SortingContext;
import java.util.Scanner;

/**
 *
 * @author Micaela Jacome DESKTOP-46VMNHU ESPE
 */
public class SortApp {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        SortingContext sortingContext = new SortingContext();
        
        System.out.print("Ingrese el numero de elementos: ");
        int n = scanner.nextInt();
        
        int[] data = new int[n];
        System.out.println("Ingrese " + n + " numeros:");
        
        for (int i = 0; i < n; i++) {
            data[i] = scanner.nextInt();
        }
        
        System.out.println("Array original:");
        printArray(data);
        
        int[] sortedData = sortingContext.sort(data.clone());
        
        System.out.println("Array ordenado:");
        printArray(sortedData);
        
        scanner.close();
    }
    
    private static void printArray(int[] array) {
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.println();
    }
}

File: hw/astudillo/U3/HW21Strategy/StrategyPython/view/sort_app.py
import sys
import os
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from controller.sorting_context import SortingContext

class SortApp:
    
    @staticmethod
    def main():
        sorting_context = SortingContext()
        
        print("Ingrese el numero de elementos: ", end="")
        n = int(input())
        
        data = []
        print(f"Ingrese {n} numeros:")
        
        for i in range(n):
            data.append(int(input()))
        
        print("Array original:")
        SortApp.print_array(data)
        
        sorted_data = sorting_context.sort(data.copy())
        
        print("Array ordenado:")
        SortApp.print_array(sorted_data)
    
    @staticmethod
    def print_array(array):
        for num in array:
            print(num, end=" ")
        print()

if __name__ == "__main__":
    SortApp.main()
